source/common/network/io_socket_handle_impl.cc

```diff
@@ -63,6 +64,14 @@ constexpr int messageTruncatedOption() {
 
 namespace Network {
 
+bool IoSocketHandleImpl::alwaysUseV6OnAndroid() {
```

Comment:
if the end goal after runtime guards are gone is to return true on android and false elsewhere maybe forceV6ForAndroid()